sycl/source/detail/handler_impl.hpp

```diff
@@ -65,6 +65,12 @@ class handler_impl {
   /// equal to the queue associated with the handler if the corresponding
   /// submission is a fallback from a previous submission.
   std::shared_ptr<queue_impl> MSubmissionSecondaryQueue;
+
+  // Protects MAuxiliaryResources.
+  std::mutex MAuxiliaryResourcesMutex;
```

Comment:
Is this really needed? Are there any chance of handler-related operations to be multi-threaded?

Reply:
It has been removed.